### PR TITLE
[SPARK-40471][BUILD] Upgrade RoaringBitmap to 0.9.32

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims upgrade RoaringBitmap 0.9.32

### Why are the changes needed?
This is a bug fix version:

- https://github.com/RoaringBitmap/RoaringBitmap/issues/575
- https://github.com/RoaringBitmap/RoaringBitmap/pull/578

other changes as follows:

- https://github.com/RoaringBitmap/RoaringBitmap/compare/0.9.31...0.9.32

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
will check MapStatusesConvertBenchmark result later

File: core/benchmarks/MapStatusesConvertBenchmark-results.txt [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 
-OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1031-azure
+OpenJDK 64-Bit Server VM 1.8.0_345-b01 on Linux 5.15.0-1019-azure
 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
 MapStatuses Convert:                      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Num Maps: 50000 Fetch partitions:500               1001           1033          36          0.0  1000638934.0       1.0X
-Num Maps: 50000 Fetch partitions:1000              1699           1705           7          0.0  1699358972.0       0.6X
-Num Maps: 50000 Fetch partitions:1500              2647           2855         314          0.0  2646904255.0       0.4X
+Num Maps: 50000 Fetch partitions:500               1179           1187          13          0.0  1178581948.0       1.0X
```
Comment (LuciferYang) [inline core/benchmarks/MapStatusesConvertBenchmark-results.txt:9]: Let me double check the base data

Comment (LuciferYang) [reply core/benchmarks/MapStatusesConvertBenchmark-results.txt:9]: run this bench without this pr

```
================================================================================================
MapStatuses Convert Benchmark
================================================================================================

OpenJDK 64-Bit Server VM 1.8.0_345-b01 on Linux 5.15.0-1019-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
MapStatuses Convert:                      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Num Maps: 50000 Fetch partitions:500               1150           1156           5          0.0  1149981444.0       1.0X
Num Maps: 50000 Fetch partitions:1000              2304           2322          18          0.0  2304318021.0       0.5X
Num Maps: 50000 Fetch partitions:1500              3557           3792         377          0.0  3556704966.0       0.3X
```

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LuciferYang and @HyukjinKwon .
Merged to master.

**Pull request closed without merging**